packages/iov-core/src/jsonrpcsigningserver.ts

```diff
@@ -0,0 +1,184 @@
+import { ChainId, UnsignedTransaction } from "@iov/bcp-types";
+import {
+  isJsonCompatibleDictionary,
+  jsonRpcCodeInvalidParams,
+  jsonRpcCodeInvalidRequest,
+  jsonRpcCodeMethodNotFound,
+  jsonRpcCodeServerErrorDefault,
+  JsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  parseJsonRpcId,
+  parseJsonRpcRequest,
+} from "@iov/jsonrpc";
+
+import { SigningServerCore } from "./signingservercore";
+
+interface RpcCallGetIdentities {
+  readonly name: "getIdentities";
+  readonly reason: string;
+  readonly chainIds: ReadonlyArray<ChainId>;
+}
+
+interface RpcCallSignAndPost {
+  readonly name: "signAndPost";
+  readonly reason: string;
+  readonly transaction: UnsignedTransaction;
+}
+
+type RpcCall = RpcCallGetIdentities | RpcCallSignAndPost;
+
+class ParamsError extends Error {}
+class MethodNotFoundError extends Error {}
+
+function isArrayOfStrings(array: ReadonlyArray<any>): array is ReadonlyArray<string> {
+  return array.every(element => typeof element === "string");
+}
+
+function parseRpcCall(data: JsonRpcRequest): RpcCall {
+  if (!isJsonCompatibleDictionary(data.params)) {
+    throw new Error("Request params are only supported as dictionary");
+  }
+
+  switch (data.method) {
+    case "getIdentities": {
+      const reason = data.params.reason;
```

Comment:
style: 
`const { chainIds, reason } = data.params` 
is simpler and easier to read (for me).

Reply:
Jap, you are right, it's a little nicer: https://github.com/iov-one/iov-core/pull/654/commits/2c26a417ab0e644f12e056771b12e3f5ecdd9601